clientv3/watch.go

```diff
@@ -268,6 +268,13 @@ func (w *watcher) Watch(ctx context.Context, key string, opts ...OpOption) Watch
 	case reqc <- wr:
 		ok = true
 	case <-wr.ctx.Done():
+		// tear down stream if this request is the only one using it
+		wgs.mu.Lock()
+		if len(wgs.streams) == 0 && wgs.stopc != nil {
```

Comment:
can we make this a method on wgs? I remember we did the same thing in wsg.closestream?